nbgrader2canvas.py

```diff
@@ -2,7 +2,40 @@
 import glob
 import tempfile
 import os
+import hashlib
+import json
+import pandas as pd
 
+# globals
+UPLOAD_LOG = "upload_log.csv"
+UPLOAD_LOG_COLUMNS = ["canvas_id", "assignment_name", "file_upload_hash"]
+
+# functions
+def open_upload_log():
+    try:
+        upload_history = pd.read_csv(UPLOAD_LOG)
+    except FileNotFoundError:
+        upload_history = pd.DataFrame(columns=UPLOAD_LOG_COLUMNS)
+    return upload_history.set_index(UPLOAD_LOG_COLUMNS[0:2])
+
+def parse_notebook_as_json(path):
+    with open(path) as file:
+        json_dict = json.load(file)
+
+    for index, cell_dict in enumerate(json_dict["cells"]):
+        # complex outputs can slow down hashing, and could be random
+        if "outputs" in cell_dict.keys():
+            json_dict["cells"][index].pop("outputs")
+        # time of execution hampers the comparison
+        if "execution" in cell_dict["metadata"].keys():
+            json_dict["cells"][index]["metadata"].pop("execution")
+
+    return json_dict
+
+def sha256(string):
+    return hashlib.sha256(string.encode("utf-8")).hexdigest()
```

Comment:
You mentioned that this is only used once and probably doesn't need to be a function, but I feel that using 2 extra lines is a good trade off for making it much easier to read.